prusti-viper/src/encoder/procedure_encoder.rs

```diff
@@ -2164,16 +2169,49 @@ impl<'p, 'v: 'p, 'tcx: 'v> ProcedureEncoder<'p, 'v, 'tcx> {
                             debug!("Encoding call of PartialEq::ne");
                             stmts.extend(
                                 self.encode_cmp_function_call(
-                                    def_id,
-                                    location,
-                                    term.source_info.span,
-                                    args,
-                                    destination,
-                                    vir::BinOpKind::NeCmp,
+                                def_id,
+                                location,
+                                term.source_info.span,
+                                args,
+                                destination,
+                                vir::BinOpKind::NeCmp,
                                 )?
                             );
                         }
 
+                        "core::f32::<impl f32>::is_nan" |
+                        "core::f64::<impl f64>::is_nan" => {
+                            let span = self.mir_encoder.get_span_of_location(location);
+                            let operand = self.mir_encoder.encode_operand_expr(&args[0]).with_span(span)?;
+                            let operand_box = std::boxed::Box::from(operand);
+                            let expr = vir::Expr::UnaryOp(UnaryOpKind::IsNaN, operand_box, vir::Position::default());
+
+                            let label = self.cfg_method.get_fresh_label_name();
+                            stmts.push(vir::Stmt::Label(label.clone()));
+
+                            // Havoc the content of the lhs
+                            let (target_place, pre_stmts) = self.encode_pure_function_call_lhs_place(destination);
+                            stmts.extend(pre_stmts);
+                            stmts.extend(self.encode_havoc(&target_place));
+                            let type_predicate = self
+                                .mir_encoder
+                                .encode_place_predicate_permission(target_place.clone(), vir::PermAmount::Write)
+                                .unwrap();
+
+                            stmts.push(vir::Stmt::Inhale(
+                                type_predicate,
+                            ));
+
+                            // Store a label for permissions got back from the call
+                            self.label_after_location.insert(location, label.clone());
+
+                            let lhs = vir::Expr::Field(std::boxed::Box::from(target_place.clone()), vir::Field::new("val_bool", vir::Type::Bool), vir::Position::default());
```

Comment:
```suggestion
                            let lhs = vir::Expr::field(target_place.clone(), vir::Field::new("val_bool", vir::Type::Bool));
```
(I also think we have something in encoder for fields so they are not created directly like this?)